dango/testing/tests/dex.rs

```diff
@@ -2187,6 +2238,103 @@ fn swap_exact_amount_out(
+    suite
+        .execute(
+            &mut accounts.owner,
+            contracts.dex,
+            &dex::ExecuteMsg::ProvideLiquidity {
+                base_denom: dango::DENOM.clone(),
+                quote_denom: usdc::DENOM.clone(),
+            },
+            coins! {
+                dango::DENOM.clone() => 1000000,
+                usdc::DENOM.clone() => 1000000,
+            },
+        )
+        .should_succeed();
+
+    // Update pair params to change pool type to geometric
+    suite
+        .query_wasm_smart(contracts.dex, dex::QueryPairRequest {
+            base_denom: dango::DENOM.clone(),
+            quote_denom: usdc::DENOM.clone(),
+        })
+        .should_succeed_and(|pair_params: &PairParams| {
+            // Update pair params
+            suite
+                .execute(
+                    &mut accounts.owner,
+                    contracts.dex,
+                    &dex::ExecuteMsg::BatchUpdatePairs(vec![PairUpdate {
+                        base_denom: dango::DENOM.clone(),
+                        quote_denom: usdc::DENOM.clone(),
+                        params: PairParams {
+                            lp_denom: pair_params.lp_denom.clone(),
+                            swap_fee_rate: pair_params.swap_fee_rate,
+                            pool_type: PassiveLiquidity::Geometric {
+                                order_spacing: Udec128::ONE,
+                                ratio: Bounded::new_unchecked(Udec128::ONE),
+                            },
+                        },
+                    }]),
+                    Coins::new(),
+                )
+                .should_succeed();
+            true
+        });
+
+    // Ensure swap exact amount in fails
+    suite
+        .execute(
+            &mut accounts.user1,
+            contracts.dex,
+            &dex::ExecuteMsg::SwapExactAmountIn {
+                route: MaxLength::new_unchecked(
+                    UniqueVec::try_from(vec![PairId {
+                        base_denom: dango::DENOM.clone(),
+                        quote_denom: usdc::DENOM.clone(),
+                    }])
+                    .unwrap(),
+                ),
+                minimum_output: None,
+            },
+            coins! {
+                dango::DENOM.clone() => 1000000,
+            },
+        )
+        .should_fail_with_error(
+            "data not found! type: dango_types::oracle::price_source::PriceSource",
+        );
+
+    // Ensure swap exact amount out fails
+    suite
+        .execute(
+            &mut accounts.user1,
+            contracts.dex,
+            &dex::ExecuteMsg::SwapExactAmountOut {
+                route: MaxLength::new_unchecked(
+                    UniqueVec::try_from(vec![PairId {
+                        base_denom: dango::DENOM.clone(),
+                        quote_denom: usdc::DENOM.clone(),
+                    }])
+                    .unwrap(),
```

Comment:
You can just do `UniqueVec::new_unchecked` if you already know the elements are unique.